tuning/tuning.md

```diff
@@ -0,0 +1,315 @@
+
+##### Prebuff-Sort-On-Consume
+
+Example: `Prebuff-Sort-On-Consume=true`
+
+The Prebuff-Sort-On-Consume parameter tells the prebuffer to sort locks of data prior to pushing them to disk. The sorting process is only applied to the individual block, and does NOT guarantee that data is sorted when entering the pipeline. Sorting blocks prior to storage also incurs a significant performance penalty in ingestion. Almost all installations should leave this value as false.
+
+##### Max-Block-Size
+
+Example: `Max-Block-Size=8`
+
+The Max-Block-Size specifies a value in megabytes and is used as a hint to tell indexers the maximum block size they can generate when pushing entries into the pipeline. Larger blocks reduce pressure on the pipeline, but increase memory pressure. Large memory and high throughput systems can increase this value to increase throughput, smaller memory systems can decrease this size to reduce memory pressure. The Prebuff-Block-Hint and Max-Block-Size parameters intersect to provide two knobs that tune ingest and search throughput. At Gravwell, on the 128GB nodes, the following is achieved: a clean 1GB/s of search throughput; a 1.25 million entry per second ingest with a Max-Block-Size of 16; and a Prebuff-Block-Hint of 8 is achieved.
+
+#### Well configuration options
+
+##### Disable-Compression, Disable-Hot-Compression, Disable-Cold-Compression
+
+Example: `Disable-Compression=true`
+
+These parameters control user-mode compression of data in the wells. By default, Gravwell will compress data in the well. Setting `Disable-Hot-Compression` or `Disable-Cold-Compression` will disable it for the hot or cold storage, respectively; setting `Disable-Compression` disables it for both.
+
+##### Enable-Transparent-Compression, Enable-Hot-Transparent-Compression, Enable-Cold-Transparent-Compression
+
+Example: `Enable-Transparent-Compression=true`
+
+These parameters control kernel-level, transparent compression of data in the wells. If enabled, Gravwell can instruct the `btrfs` filesystem to transparently compress data. This is more efficient than user-mode compression. Setting `Enable-Transparent-Compression` true automatically turns off user-mode compression. Note that setting `Disable-Compression=true` will disable transparent compression.
+
+##### Acceleration
+
+As mentioned above, utilizing well acceleration can dramatically improve performance. See the [Acceleration documentation](#!configuration/accelerators.md) for more information.
+
+### Linux Kernel and System Tuning
+
+#### Disable swap
+
+If possible, disable swap support on any Gravwell indexer node. Swap helps alleviate memory pressure at the expense of writing data to disk. A busy Gravwell indexer is likely to create _thrashing_ in the swap subsystem. 
```

Comment:
is this true?  Is an OOM better than a slowdown?

Reply:
I just removed this recommendation. A real recommendation would be to adjust vm.swapiness, but that has so many caveats...